tests/cc-stale.mjs

```diff
@@ -0,0 +1,112 @@
+
+export default {
+  name: 'Cache-Control Extensions for Stale Content',
+  id: 'cc-stale',
+  description: '[RFC5861](https://httpwg.org/specs/rfc5861.html) defines two independent HTTP Cache-Control extensions that allow control over the use of stale responses by caches. These tests gather information about whether implementations honor them by default.',
+  tests: [
+    {
+      name: 'An optimal cache serves cached `stale-while-revalidate` responses while revalidating in the background',
+      id: 'stale-while-revalidate',
+      kind: 'optimal',
+      requests: [
+        {
+          setup: true,
+          response_headers: [
+            ['Cache-Control', 'max-age=0, stale-while-revalidate=3600'],
+            ['ETag', '"abc"']
+          ]
+        },
+        {
+          pause_after: true,
+          expected_response_headers: [
+            ['Cache-Control', 'max-age=0, stale-while-revalidate=3600'],
+            ['ETag', '"abc"']
+          ],
+          response_headers: [
+            ['Cache-Control', 'max-age=3600', false],
+            ['ETag', '"def"', false]
+          ]
+        },
+        {
+          expected_type: 'cached',
```

Comment:
I don't think this is going to work; the check for `cached` presumes that the request was created by the client (tracked using the `Req-Num` request header); here, the request is created by the cache itself.